crates/primitives/starknet/src/execution/felt252_wrapper.rs

```diff
@@ -0,0 +1,373 @@
+//! # Felt252 - FieldElement wrapper.
+//!
+//! Starknet base type is a [`FieldElement`] from starknet-ff crate.
+//! Substrate primitives are passed back and forth between client
+//! and runtime using SCALE encoding: https://docs.substrate.io/reference/scale-codec/.
+//!
+//! The [`Felt252Wrapper`] implements the traits for SCALE encoding, and wrap
+//! the [`FieldElement`] type from starknet-ff.
+
+use core::ops::Deref;
+
+use cairo_vm::felt::Felt252;
+use scale_codec::{Decode, Encode, EncodeLike, Error, Input, MaxEncodedLen, Output};
+use scale_info::build::Fields;
+use scale_info::{Path, Type, TypeInfo};
+use sp_core::{H256, U256};
+use starknet_ff::{FieldElement, FromByteSliceError, FromStrError};
+
+///
+#[derive(Clone, Debug, PartialEq, Eq, Copy)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+pub struct Felt252Wrapper(pub FieldElement);
+
+impl Felt252Wrapper {
+    /// Inits from '0x...' hex string.
+    pub fn from_hex_be(value: &str) -> Result<Self, Felt252WrapperError> {
+        let ff = FieldElement::from_hex_be(value)?;
+        Ok(Self(ff))
+    }
+
+    /// Inits from decimal string.
+    pub fn from_dec_str(value: &str) -> Result<Self, Felt252WrapperError> {
+        let ff = FieldElement::from_dec_str(value)?;
+        Ok(Self(ff))
+    }
+
+    /// Inits from zero constant value.
+    pub fn zero() -> Felt252Wrapper {
```

Comment:
I see that you use `Deref` below, so the only purposed of these is to change the case and type? (noob question)

Reply:
Tried to remove it but couldn't access inner type public constants, not sure why.

Reply:
@tdelabro an idea?

Reply:
https://doc.rust-lang.org/std/ops/trait.Deref.html

```
Implementing Deref for smart pointers makes accessing the data behind them convenient, which is why they implement Deref. On the other hand, the rules regarding Deref and [DerefMut](https://doc.rust-lang.org/std/ops/trait.DerefMut.html) were designed specifically to accommodate smart pointers. Because of this, Deref should only be implemented for smart pointers to avoid confusion.
```

We should not use deref just to access the inner type of a wrapper